src/tracer.js

```diff
@@ -80,6 +80,9 @@ export default class Tracer {
             } else {
                 fields.operationName = nameOrFields;
             }
+            if (fields.parent) {
+                fields.parent = fields.parent._imp;
```

Comment:
I could argue the opposite, that implementation should always handle the arguments of the _interface it implements_, and convert them to internal implementation as needed.

That's not a sticking point to me, but the fact that we're changing the dictionary passed as argument is. I'd prefer an API that does not have side effects, so the following would've made more sense to me:

```javascript
startSpan(operationName, fields) {
  if (API_CONFORMANCE_CHECKS) { ... }
  return new Span(this._imp.startSpan(operationName, fields));
}
```